### PR TITLE
Remove deprecated config USE_SINGLE_REGION

### PR DESCRIPTION
**Pull request by viren-nadkarni**
## Motivation

The `USE_SINGLE_REGION` environment config variable was used in early years of LocalStack when multi-region was unsupported. It has since been deprecated.

This PR removes this config variable for the upcoming major release.

### CODE REVIEW

File: localstack/runtime/analytics.py [hunk 1]
```diff
@@ -37,7 +37,6 @@
     "REQUIRE_PRO",
     "SKIP_INFRA_DOWNLOADS",
     "SQS_ENDPOINT_STRATEGY",
-    "USE_SINGLE_REGION",
```
Comment (alexrashed) [inline localstack/runtime/analytics.py:40]: I think we should still keep track of the usage of the old variables (this way we can reason if it's still trying to be used, even though it does not have any effects anymore).

File: localstack/deprecations.py [hunk 2]
```diff
@@ -47,12 +47,6 @@ def is_affected(self) -> bool:
         "0.12.8",
         "PORT_WEB_UI has been removed, and is not available anymore. Please remove this environment variable.",
     ),
-    # Since 0.12.7 - Full multi-region support
-    EnvVarDeprecation(
-        "USE_SINGLE_REGION",
-        "0.12.7",
-        "LocalStack now has full multi-region support. Please remove this environment variable.",
-    ),
```
Comment (alexrashed) [inline localstack/deprecations.py:55]: I think we should keep this deprecation warning, but just slightly change the text to something like "LocalStack now has full multi-region support.This variable has been removed and does not have any effect anymore. Please remove it form your configuration."

File: localstack/config.py [hunk 3]
```diff
@@ -1286,7 +1283,6 @@ def use_custom_dns():
     "SYNCHRONOUS_SNS_EVENTS",
     "TEST_AWS_ACCOUNT_ID",
     "TF_COMPAT_MODE",
-    "USE_SINGLE_REGION",
```
Comment (alexrashed) [inline localstack/config.py:1289]: I think we should still forward the variable in order to make sure that we show the deprecation warnings / inform the user that it's not having any effect anymore.

**Review by alexrashed — COMMENTED** [review]
Great to see that we are finally getting rid of the old deprecated variables. 🧹 
However, I think we should still show warnings to users who are still using it...

File: localstack/config.py (hunk 3 above)
Comment (viren-nadkarni) [reply localstack/config.py:1286]: Good idea, fixed in 46efde0a7

File: localstack/runtime/analytics.py (hunk 1 above)
Comment (viren-nadkarni) [reply localstack/runtime/analytics.py:40]: Restored in 46efde0a7

**Review by alexrashed — APPROVED** [review]
Aweseome! Thanks for addressing the comments and bringing the removal of the `USE_SINGLE_REGION` over the line! :rocket:

**Pull request merged**